src/main/java/com/alipay/oceanbase/rpc/stream/ObTableClientQueryStreamResult.java

```diff
@@ -76,10 +80,20 @@ protected ObTableQueryResult execute(ObPair<Long, ObTableParam> partIdWithIndex,
                         if (failedServerList != null) {
                             route.setBlackList(failedServerList);
                         }
-                        subObTable = client
-                            .getTable(indexTableName, partIdWithIndex.getLeft(),
-                                needRefreshTableEntry, client.isTableEntryRefreshIntervalWait(),
-                                route).getRight().getObTable();
+                        if (isNeedRefreshTableGroupName) {
+                            // if input is table group, retry should replace table name
+                            indexTableName = client.tryGetTableNameFromTableGroupCache(client.getTableGroupInverted().get(indexTableName), true);
+                            tableName = indexTableName;
+                        }
+                        tableParam = client.getTable(indexTableName, partIdWithIndex.getLeft(),
+                                        needRefreshTableEntry, client.isTableEntryRefreshIntervalWait(),
+                                        route).getRight();
+                        subObTable = tableParam.getObTable();
+                        if (isNeedRefreshTableGroupName) {
```

Comment:
Reset isNeedRefreshTableGroupName
Also, whether it needs to refresh the tableGroup name, we could set new tableId and tabletId. This means we could remove `if`

Reply:
edit , delete retry get table name